src/Pagination.jsx

```diff
@@ -0,0 +1,133 @@
+import React, { Component } from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+
+export class Pagination extends Component {
+  state = {
+    selectedPage: this.props.page,
+    totalPageCount: Math.ceil(this.props.total / this.props.perPage),
+    firstPage: this.props.page,
+    data: this.props.data,
+  }
+
+  setNumber = pageNumber => () => {
+    this.setState({
+      selectedPage: pageNumber,
+    });
+  }
+
+  prevNumber = () => {
+    if (this.state.selectedPage <= this.state.firstPage) {
+      return;
+    }
+
+    this.setState(prevState => ({
+      selectedPage: prevState.selectedPage - 1,
+    }));
+  }
+
+  nextNumber = () => {
+    if (this.state.selectedPage >= this.state.totalPageCount) {
+      return;
+    }
+
+    this.setState(prevState => ({
+      selectedPage: prevState.selectedPage + 1,
+    }));
+  }
+
+  render() {
+    const { selectedPage, data } = this.state;
+    const { perPage } = this.props;
+    const { totalPageCount } = this.state;
```

Comment:
не хочешь сразу создать все переменные? 
```suggestion
    const {
     selectedPage, 
    data, 
    totalPageCount,
    } = this.state;
    const { perPage } = this.props;
```